docs/topics/proc_exposing-application-metrics-using-prometheus-with-vertx.adoc

```diff
@@ -105,54 +108,43 @@ import io.vertx.micrometer.backends.BackendRegistries;
 
 public class MyVertxApp extends AbstractVerticle {
 
+  public class CustomLauncher extends Launcher {
+
+    @Override
+    public void beforeStartingVertx(VertxOptions options) {
+      options.setMetricsOptions(new MicrometerMetricsOptions()
+        .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
+          .setStartEmbeddedServer(true)
+          .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
+          .setEmbeddedServerEndpoint("/metrics"))
+        .setEnabled(true));
+    }
+
     @Override
-    public void start() {
-        Vertx vertx = Vertx.vertx(new VertxOptions().setMetricsOptions(
-            new MicrometerMetricsOptions()
-                .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
-                    .setStartEmbeddedServer(true)
-                    .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
-                    .setEmbeddedServerEndpoint("/metrics"))
-                .setEnabled(true)));
-
-        PrometheusMeterRegistry registry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
-        registry.config().meterFilter(
-            new MeterFilter() {
-                @Override
-                public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
-                return DistributionStatisticConfig.builder()
-                    .percentilesHistogram(true)
-                    .build()
-                    .merge(config);
-                }
-            });
-
-        vertx.deployVerticle(MyVertxApp.class.getName());
-
-        Timer timer = Timer
-            .builder("my.timer")
-            .description("a description of what this timer does")
-            .register(registry);
-
-        vertx.setPeriodic(1000, l -> {
-            timer.record(() -> {
-                vertx.createHttpServer()
-                .requestHandler(req -> req.response().end("Hello"))
-                .listen(8080);
-            });
+    public void afterStartingVertx(Vertx vertx) {
+      PrometheusMeterRegistry registry = (PrometheusMeterRegistry) BackendRegistries.getDefaultNow();
+      registry.config().meterFilter(
+        new MeterFilter() {
+          @Override
+          public DistributionStatisticConfig configure(Meter.Id id, DistributionStatisticConfig config) {
+            return DistributionStatisticConfig.builder()
+              .percentilesHistogram(true)
+              .build()
+              .merge(config);
+          }
         });
     }
-}
+  }
 ----
 --
-
-. Set the `<vertx.verticle>` property in the `pom.xml` file of your application to your custom class:
+. Set the `<vertx.verticle>` and `<vertx.launcher>` properties in the `pom.xml` file of your application.
 +
 [source,xml,options="nowrap"]
 ----
 <properties>
   ...
   <vertx.verticle>org.acme.MyVertxApp</vertx.verticle>
+  <vertx.launcher>CustomLauncher</vertx.launcher>
```

Comment:
This must be a full qualified class name like `vertx.verticle`

Reply:
What is a typical example of the path for the launcher?

Reply:
`org.acme.CustomLauncher`. Do not forget the package declaration in the classes (verticle and launcher)